modules/core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryVersioning.java

```diff
@@ -1071,4 +1072,21 @@ public void testDirtyStateBehaviours() {
         assertNotEquals(v3, doc.getVersionLabel());
     }
 
+    // NXP-30075
+    @Test
+    public void testVersionNotModifiableException() {
+        DocumentModel doc = session.createDocumentModel("/", "doc", "File");
+        doc = session.createDocument(doc);
+        session.save();
+        DocumentRef vRef = session.checkIn(doc.getRef(), VersioningOption.MAJOR, null);
+        DocumentModel version = session.getDocument(vRef);
+
+        try {
+            session.checkIn(vRef, VersioningOption.MAJOR, null);
```

Comment:
Why doing a check in instead of simply updating and saving the doc?

Reply:
Updating and saving the document throws a `PropertyException`.